### PR TITLE
Add support for RSA X509 certificates

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Add support for RSA X509 certificates.
Try to keep the main certificate interface generic and hide the different key and signature types in enums `PublicKey` and `Signature`.

### Motivation

In pulling in the tests from https://csrc.nist.gov/projects/pki-testing it was realized that an assumption was made that the certificates would always be ECDSA

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #68** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/68" target="_blank">Graphite</a> 
    * **PR #69** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/69" target="_blank">Graphite</a> 
      * **PR #70** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/70" target="_blank">Graphite</a> 
        * **PR #71** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/71" target="_blank">Graphite</a> 
          * **PR #72** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/72" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/attestation/72?utm_source=stack-comment).

File: verifier/src/x509/algorithm.rs [hunk 1]
```diff
@@ -0,0 +1,444 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Algorithm data types used in x509 certificate logic
```
Comment (nick-mobilecoin) [inline verifier/src/x509/algorithm.rs:3]: Up for suggestions on this naming. "algorithm" feels a little odd of a name, this is the signature and keys. Maybe each type should go to a separate module?

File: verifier/src/x509/chain.rs [hunk 2]
```diff
@@ -260,3 +270,67 @@ mod test {
         );
     }
 }
+
+#[cfg(test)]
+mod pkits {
```
Comment (nick-mobilecoin) [inline verifier/src/x509/chain.rs:275]: My thought was to separate the pkits tests from the others.

And as I write this perhaps the pkits tests belong as integration tests...???  I'm up for suggestions there.

Comment (nick-mobilecoin) [reply verifier/src/x509/chain.rs:275]: Moved to integration tests in #75

File: verifier/data/tests/pub_keys/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+The ecdsa public keys were created using OpenSSL and the following command
+lines:
+
+```console
+openssl ecparam -name prime256v1 -genkey -noout -out <private_key_file>.pem
+openssl ec -in <private_key_file>.pem -pubout -out <public_key_file>.pem
+```
+
+For the RSA keys, the following command lines were used:
```
Comment (samdealy) [inline verifier/data/tests/pub_keys/README.md:14]:
```suggestion
For the RSA keys, the following commands were executed:
```

File: verifier/data/tests/pub_keys/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Public Keys For Testing
+
+Public keys created with openssl. The private keys are not included to avoid
+being flagged by security scans.
+
+The ecdsa public keys were created using OpenSSL and the following command
+lines:
```
Comment (samdealy) [inline verifier/data/tests/pub_keys/README.md:7]:
```suggestion
The ecdsa public keys were created using OpenSSL and the following commands:
```

**Review by samdealy — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
punting on implementing x509 chain parsing logic.

**Pull request closed without merging**